lib/packExternalModules.js

```diff
@@ -291,20 +291,23 @@ module.exports = {
         addModulesToPackageJson(prodModules, modulePackage);
         this.serverless.utils.writeFileSync(modulePackageJson, JSON.stringify(modulePackage, null, 2));
 
-        // Copy modules
-        const startCopy = _.now();
-        return BbPromise.fromCallback(callback => fse.copy(path.join(compositeModulePath, 'node_modules'), path.join(modulePath, 'node_modules'), callback))
-        .tap(() => this.options.verbose && this.serverless.cli.log(`Copy modules: ${modulePath} [${_.now() - startCopy} ms]`))
-        .then(() => {
-          // Prune extraneous packages - removes not needed ones
-          const startPrune = _.now();
-          return BbPromise.fromCallback(callback => {
-            childProcess.exec('npm prune', {
-              cwd: modulePath
-            }, callback);
-          })
-          .tap(() => this.options.verbose && this.serverless.cli.log(`Prune: ${modulePath} [${_.now() - startPrune} ms]`));
-        });
+        // Copy modules only if not google-cloud-functions
+        // GCF Auto installs the package json
+        if (_.get(this.serverless, 'service.provider.name') !== 'google') {
```

Comment:
I would prefer an early exit here if the provider is google instead of putting the default case into the if.
```
if (_.get(this.serverless, 'service.provider.name') === 'google') {
  return BbPromise.resolve();
}
...
```

Reply:
Updated this. I will add unit tests for the new behavior to restore coverage.